packages/apollo-link-ws/src/webSocketLink.ts

```diff
@@ -35,14 +35,14 @@ export class WebSocketLink extends ApolloLink {
   ) {
     super();
 
-    if (paramsOrClient instanceof SubscriptionClient) {
-      this.subscriptionClient = paramsOrClient;
-    } else {
+    if ((<WebSocketLink.Configuration>paramsOrClient).uri !== undefined) {
```

Comment:
Isn't it possible to do this in one step? Like cast it once and not every time you use it? Just asking

Reply:
Hi Jovi! Glad you're reviewing this. First of all, just to be clear: I'm not a TypeScript expert, neither a TS user... This is one of my first times writing TS code. 😄
Regarding your question: I'm not sure if we can do that... neither if it would change the real JS output, which is, right now:

```js
var WebSocketLink = (function (_super) {
    __extends(WebSocketLink, _super);
    function WebSocketLink(paramsOrClient) {
        var _this = _super.call(this) || this;
        if (paramsOrClient.uri !== undefined) {
            _this.subscriptionClient = new subscriptions_transport_ws_1.SubscriptionClient(paramsOrClient.uri, paramsOrClient.options, paramsOrClient.webSocketImpl);
        }
        else {
            _this.subscriptionClient = paramsOrClient;
        }
        return _this;
    }
    WebSocketLink.prototype.request = function (operation) {
        return this.subscriptionClient.request(operation);
    };
    return WebSocketLink;
}(apollo_link_1.ApolloLink));
```

However, if you can confirm that it could be changed, I'd change it indeed. Also feel free to change anything!

Reply:
I'll try to free up some time this afternoon to look at it, I know the JS out put does not change it just reads pretty hard casting it multiple times.

Reply:
In vanilla JS I'd probably write something like this:

```ts
if ((<WebSocketLink.Configuration>paramsOrClient).uri) {
  const { uri, options, webSocketImpl } = (<WebSocketLink.Configuration>paramsOrClient);
  this.subscriptionClient = new SubscriptionClient(uri, options, webSocketImpl);
} else {
  this.subscriptionClient = <SubscriptionClient>paramsOrClient;
}
```

I've tried it and it works / compile as expected. Could work for you?

Reply:
Yes, I'll take it up to the apollo-maintainers tomorrow. Thanks for the quick feedback